src/connection-manager/auto-dialler.ts

```diff
@@ -137,14 +150,29 @@ export class AutoDialler implements Startable {
       }
 
       const peer = peers[i]
+      const peerIdString = peer.id.toString()
 
-      if (this.components.connectionManager.getConnections(peer.id).length === 0) {
+      if (this.components.connectionManager.getConnections(peer.id).length === 0 && !this.connectionMap.has(peerIdString)) {
         log('connecting to a peerStore stored peer %p', peer.id)
-        try {
-          await this.components.connectionManager.openConnection(peer.id)
-        } catch (err: any) {
+        const connectionAbortController = new AbortAbort()
+        this.abortController?.addDependant(connectionAbortController)
+
+        /**
+         * we do not need to await the connection attempts because we are
+         * blocking new calls to `openConnection` by checking whether the
+         * peer already has an attempt in the `connectionMap` and we are
+         * removing the peer from the `connectionMap` when the attempt
+         * finishes (successful or not)
+         */
+        const connectionAttempt = this.components.connectionManager.openConnection(peer.id, { signal: connectionAbortController.signal })
```

Comment:
`peers` contains every peer in the peer store - I think as written this will dial every peer ever encountered when the number of connections drops below the threshold, potentially swamping the dial queue.

This needs to have a configurable bound by putting the `connectionManager .openConnection` calls in `p-queue` items or something and exposing the `concurrency` option as config for the connection manager.

Reply:
> peers contains every peer in the peer store - I think as written this will dial every peer ever encountered when the number of connections drops below the threshold, potentially swamping the dial queue.

That makes sense, thanks for the context. see https://github.com/libp2p/js-libp2p/pull/1668#issuecomment-1494741602

Reply:
> This needs to have a configurable bound by putting the connectionManager.openConnection calls in p-queue items or something and exposing the concurrency option as config for the connection manager.

I imagine we would want to do that in this PR. I will look into it